src/kixi/comms/components/kafka.clj

```diff
@@ -192,31 +203,33 @@
   (async/thread
     (try
       (let [^FranzConsumer consumer (consumer/make-consumer
-                                     config
+                                     (dissoc-custom config)
                                      (deserializers/string-deserializer)
                                      (deserializers/string-deserializer)
                                      (consumer-options))
             _ (cp/subscribe-to-partitions! consumer (:consumer.topic config))]
         (loop []
-          (let [pack (cp/poll! consumer)]      
-            (cp/pause! consumer (cp/assigned-partitions consumer))
-            (doseq [raw-msg (into [] pack)]
-              (when-let [msg (some-> raw-msg
-                                     :value
-                                     transit->clj)]
-                (when (process-msg?-fn msg)            
-                  (let [result-ch (msg-handler raw-msg msg)]
-                    (loop []
-                      (let [[val port] (async/alts!! [result-ch
-                                                      (async/timeout (:heartbeat.interval.send.ms config))])]
-                        (when-not (= port
-                                     result-ch)
-                          (cp/poll! consumer)
-                          (recur))))))))
-            (cp/resume! consumer (cp/assigned-partitions consumer))
+          (let [pack (cp/poll! consumer)]
```

Comment:
The pause/resume thing? Yeah, they've made the mistake of making poll both the message fetcher and heartbeater. It really has this thing about messages being fast to process kinda baked in.